src/panels/ConfigPanel/ConfigPanel.tsx

```diff
@@ -215,29 +220,62 @@ export default function ConfigPanel(): JSX.Element {
   function generateConfirmationDialog(): JSX.Element | null {
     if (confirmType && appName) {
       const changedConfigList = generateChangedKeyValues(config);
-
-      if (confirmType === "apply") {
-        return SaveConfirmation(
-          appName,
-          changedConfigList,
-          () => {
-            setConfirmType(null);
-            // Clear the form errors if there were any from a previous submit.
-            setFormErrors(null);
-            _submitToJuju();
-          },
-          () => setConfirmType(null)
+      if (confirmType === "submit") {
+        // Render the submit confirmation modal.
+        return (
+          <Portal>
```

Comment:
This same change also needs to happen in `ActionsPanel`.

Reply:
Thanks for pointing this out! I think it needs to be added to `CharmActionsPanel` as well.